packages/flutter/test/material/app_test.dart

```diff
@@ -143,4 +143,21 @@ void main() {
 
     expect(find.text('Home'), findsOneWidget);
   });
+
+  testWidgets('Could not implicitly use root initialRoute',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(new MaterialApp(routes: <String, WidgetBuilder>{
+      '/': (context) => new Text('route "/"')
+    }));
+  });
+
+  testWidgets('Could not use non-root initialRoute',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(new MaterialApp(
+      initialRoute: '/a',
+      routes: <String, WidgetBuilder>{
+        '/a': (context) => new Text('route "/a"')
+      },
+    ));
+  });
 }
```

Comment:
another good test to have would be one with two routes, '/' and '/a', and verify that it picks the right ones when the initialRoute is not set and when it's set to '/a'. Also maybe another that tests that if you change the initialRoute, it has no effect (pumpWidget with one set of routes and an initialRoute, and then pumpWidget again in the same test with an identical set of widgets but with no initialRoute, or a different intialRoute, and verify that it doesn't change what's visible).